shared/GIMessage.js

```diff
@@ -74,7 +114,78 @@ export class GIMessage {
   serialize (): string { return this._mapping.value }
 
   hash (): string { return this._mapping.key }
-  // signWithKey (key: Uint8Array) { this._sig = nacl.sign.detached(this.toProtobuf(), key) }
-  // get signature (): Buffer { return this._sig }
-  // set signature (sig: Buffer) { this._sig = sig }
+}
+
+sbp('sbp/selectors/register', {
+  // https://www.wordnik.com/words/chelonia
+  'chelonia/message/process': function (message: GIMessage, state: Object) {
+    sanityCheck(message)
+    const [opT, opV] = message.op()
+    const hash = message.hash()
+    const contractID = message.contractID()
+    const config = sbp('okTurtles.data/get', 'CHELONIA_CONFIG')
+    if (!state._vm) state._vm = {}
+    const opFns = {
+      [GIMessage.OP_CONTRACT] (v) {
+        if (!state._vm.authorizedKeys) state._vm.authorizedKeys = []
```

Comment:
Could initialize directly without push here

Reply:
Do you mean something like this?

```js
if (!state._vm.authorizedKeys) {
  state._vm.authorizedKeys = [{ key: v.authkey, context: 'owner' }]
  return
}
```

That is a few extra lines and it would violate DRY since `{ key: v.authkey, context: 'owner' }` is repeated, so I chose to do it this way